### ISSUE TITLE
Darktable stops building on Fedora 38

### ISSUE THREAD

**Issue by jenshannoschwalm** [darktable-org/darktable#14488]
### Describe the bug

Since this morning dt stops building on Fedora 38

### Steps to reproduce

1. Use git master and build

### Logfile | Screenshot | Screencast

```
In function ‘_ioppr_generate_profile_info’,
    inlined from ‘dt_ioppr_add_profile_info_to_list’ at /home/hanno/sources/darktable/src/common/iop_profile.c:877:7,
    inlined from ‘dt_ioppr_add_profile_info_to_list’ at /home/hanno/sources/darktable/src/common/iop_profile.c:865:1:
/home/hanno/sources/darktable/src/common/iop_profile.c:780:8: error: ‘dt_colorspaces_get_matrix_from_input_profile’ accessing 64 bytes in a region of size 16 [-Werror=stringop-overflow=]
  780 |     if(dt_colorspaces_get_matrix_from_input_profile(rgb_profile, profile_info->matrix_in,
      |        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  781 |                                                     profile_info->lut_in[0],
      |                                                     ~~~~~~~~~~~~~~~~~~~~~~~~
  782 |                                                     profile_info->lut_in[1],
      |                                                     ~~~~~~~~~~~~~~~~~~~~~~~~
  783 |                                                     profile_info->lut_in[2],
      |                                                     ~~~~~~~~~~~~~~~~~~~~~~~~
  784 |                                                     profile_info->lutsize) == 0
      |                                                     ~~~~~~~~~~~~~~~~~~~~~~
compilation terminated due to -Wfatal-errors.

```

### Commit

8df1fb1

### Where did you install darktable from?

darktable.org

### darktable version

today's master

### What OS are you using?

Linux

### What is the version of your OS?

Fedora 36

### Describe your system?

gcc 13.1.1, Intel

### Are you using OpenCL GPU in darktable?

None

**Comment by jenshannoschwalm** [comment]
Thanks for hinting. Can confirm that
```
#if __GNUC__ >= 12
#define GCC12_SUPPRESS_ERRONEOUS_STRINGOP_OVERFLOW_WARNING \
```

works fine.

**Comment by Scorpion078** [comment]
i get the same error when building in win11 on two different pc

**Comment by victoryforce** [comment]
> i get the same error when building in win11 on two different pc

Please see temporary fix in the comment above. You can make this change manually yourself ot wait until it will be fixed in master. This could be the above-mentioned temporary fix that is currently part of the #14487, or the correct fix that @kmilos is preparing, whichever comes first.

**Comment by Scorpion078** [comment]
> > i get the same error when building in win11 on two different pc
> 
> Please see temporary fix in the comment above. You can make this change manually yourself ot wait until it will be fixed in master. This could be the above-mentioned temporary fix that is currently part of the #14487, or the correct fix that @kmilos is preparing, whichever comes first.

Thanks. took me little but i figured it out and it's working
